safe_grid_gym/envs/gridworlds_env.py

```diff
@@ -108,25 +108,31 @@ def step(self, action):
             if k not in ("board", "RGB"):
                 info[k] = v
 
-        if self.cheat:
-            if hidden_reward is None:
-                error.Error("gridworld '%s' does not support cheating" % self._env_name)
-                return_reward = reward
-                self.cheat = False
-            else:
-                return_reward = hidden_reward
+        board = copy.deepcopy(obs["board"])
+
+        if self._use_transitions:
+            state = np.stack([self._last_board, board], axis=0)
+            self._last_board = board
         else:
-            return_reward = reward
+            state = board
 
-        board = copy.deepcopy(obs["board"])
-        return (board, return_reward, done, info)
+        return (state, reward, done, info)
 
     def reset(self):
         timestep = self._env.reset()
         self._rgb = timestep.observation["RGB"]
         if self._viewer is not None:
             self._viewer.reset_time()
-        return timestep.observation["board"]
+
+        board = copy.deepcopy(timestep.observation["board"])
+
+        if self._use_transitions:
+            state = np.stack([board, board], axis=0)
```

Comment:
I'd suggest `state = np.stack([np.zeros(board.shape, dtype=board.dtype), board], axis=0)` here

Reply:
Yes you are right this is the better solution! But may I tell you about `np.zeros_like` :smile: